### DISCUSSION TITLE
Example webserver supporting CORS

### DISCUSSION THREAD

**Discussion by sygi** [godotengine/godot-proposals#6383] [Platforms]
With Godot 4, the web exports require the HTTP server to support CORS. This is a complication for the users, who report errors with Godot ([1](https://github.com/godotengine/godot/issues/69564), [2](https://github.com/godotengine/godot/issues/65488)). I spent a while trying to figure out a simple server and got this:

```python
import http.server
import socketserver

PORT = 8000

class Handler(http.server.SimpleHTTPRequestHandler):
    def send_response(self, *args, **kwargs):
        http.server.SimpleHTTPRequestHandler.send_response(self, *args, **kwargs)
        self.send_header('Cross-Origin-Embedder-Policy', 'require-corp')
        self.send_header('Cross-Origin-Opener-Policy', 'same-origin')

with socketserver.TCPServer(("", PORT), Handler) as httpd:
    print('serving at port', PORT)
    httpd.serve_forever()
```

It would save people a lot of time if the information on how to start a server was available directly in [the docs](https://docs.godotengine.org/en/latest/tutorials/export/exporting_for_web.html#doc-javascript-secure-contexts) or ideally, in the error itself; currently message is:
```
Warning!
The following features required to run Godot projects on the Web are missing:
Cross Origin Isolation
SharedArrayBuffer
```

which tells nothing on how to fix the issue for someone not well-versed in HTTP protocols.

**Comment by Calinou** [comment]
There's a pull request adding this information, which I just rebased and expanded upon: https://github.com/godotengine/godot-docs/pull/6108
